sys/include/net/lora.h

```diff
@@ -38,66 +40,161 @@ extern "C" {
+ * air time usage regulations before varying the SF. To calculate air time refer
+ * https://www.loratools.nl/#/airtime .
+*/
+#if IS_ACTIVE(CONFIG_LORA_SF_DEFAULT_SF6)
+#define CONFIG_LORA_SF_DEFAULT                      (LORA_SF6)
+#elif IS_ACTIVE(CONFIG_LORA_SF_DEFAULT_SF7)
+#define CONFIG_LORA_SF_DEFAULT                      (LORA_SF7)
+#elif IS_ACTIVE(CONFIG_LORA_SF_DEFAULT_SF8)
+#define CONFIG_LORA_SF_DEFAULT                      (LORA_SF8)
+#elif IS_ACTIVE(CONFIG_LORA_SF_DEFAULT_SF9)
+#define CONFIG_LORA_SF_DEFAULT                      (LORA_SF9)
+#elif IS_ACTIVE(CONFIG_LORA_SF_DEFAULT_SF10)
+#define CONFIG_LORA_SF_DEFAULT                      (LORA_SF10)
+#elif IS_ACTIVE(CONFIG_LORA_SF_DEFAULT_SF11)
+#define CONFIG_LORA_SF_DEFAULT                      (LORA_SF11)
+#elif IS_ACTIVE(CONFIG_LORA_SF_DEFAULT_SF12)
+#define CONFIG_LORA_SF_DEFAULT                      (LORA_SF12)
 #endif
 
-/** @brief Set fixed payload length on */
-#ifndef LORA_FIX_LENGTH_PAYLOAD_ON_DEFAULT
-#define LORA_FIX_LENGTH_PAYLOAD_ON_DEFAULT     (false)
+#ifndef CONFIG_LORA_SF_DEFAULT
+#define CONFIG_LORA_SF_DEFAULT                      (LORA_SF7)
 #endif
 
-/** @brief Set inverted IQ on */
-#ifndef LORA_IQ_INVERTED_DEFAULT
-#define LORA_IQ_INVERTED_DEFAULT               (false)
+/** @brief Set Coding Rate (CR)
+ *
+ * Configure Coding Rate (CR). CR denotes the implementation of forward error
+ * correction (FEC). This may be done by encoding 4-bit data with redundancies
+ * into 5-bit, 6-bit, 7-bit, or 8-bit. Coding Rate (CR) value need to be
+ * adjusted according to conditions of the channel used for data transmission.
+ * If there are too many interferences in the channel, then it’s recommended to
+ * increase the value of CR. However, the rise in CR value will also increase
+ * the duration for the transmission. Refer to country specific air time usage
+ * regulations before varying the CR. To calculate air time refer
+ * https://www.loratools.nl/#/airtime .
+*/
+#if IS_ACTIVE(CONFIG_LORA_CR_DEFAULT_CR_4_5)
+#define CONFIG_LORA_CR_DEFAULT                      (LORA_CR_4_5)
+#elif IS_ACTIVE(CONFIG_LORA_CR_DEFAULT_CR_4_6)
+#define CONFIG_LORA_CR_DEFAULT                      (LORA_CR_4_6)
+#elif IS_ACTIVE(CONFIG_LORA_CR_DEFAULT_CR_4_7)
+#define CONFIG_LORA_CR_DEFAULT                      (LORA_CR_4_7)
+#elif IS_ACTIVE(CONFIG_LORA_CR_DEFAULT_CR_4_8)
+#define CONFIG_LORA_CR_DEFAULT                      (LORA_CR_4_8)
 #endif
 
-/** @brief Frequency hopping on */
-#ifndef LORA_FREQUENCY_HOPPING_DEFAULT
-#define LORA_FREQUENCY_HOPPING_DEFAULT         (false)
+#ifndef CONFIG_LORA_CR_DEFAULT
+#define CONFIG_LORA_CR_DEFAULT                      (LORA_CR_4_5)
 #endif
 
-/** @brief Frequency hopping period */
-#ifndef LORA_FREQUENCY_HOPPING_PERIOD_DEFAULT
-#define LORA_FREQUENCY_HOPPING_PERIOD_DEFAULT  (0U)
+/** @brief Set this to 1 to enable inverted I/Q mode
+ *
+ * Inverted I/Q allows differentiating sender and receiver messages. In the case
```

Comment:
Note that this is a modulation parameter that inverts the I and Q signals. If IQ doesn't match the transceiver is not able to demodulate the LoRa frame.
In case of LoRaWAN, the gateway sends downlink frames using inverted IQ, but all uplink frames are received using non-inverted IQ. But this parameter is not LoRaWAN specific.

Thus, I propose to rephrase this to indicate what the IQ parameter means, and possibly mention the case of LoRaWAN as an example

Reply:
Yes. Intially I wrote the help keeping in mind the LoRaWAN aspect. Shall correct it